[Livia Michelle Iskandar Leo] iP

liviamil opened the pull request:
# Sally, your personal task manager bot

> "This will be done no matter what" - Livia ([img](https://images.app.goo.gl/XALxLgjZCMhucucV8))

Sally frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ **__SUPER FAST__** to use

All you need to do is,
1. download it from [here](github.com/liviamil/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [X] Managing tasks
- [X] Managing deadlines
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. here's the `main` method:
```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Some of my most used Git commands so far:
```
git status
git add
git commit
```

liviamil committed:
Level-1: Greet, Echo, Exit

liviamil committed:
Level-2: Add, List

liviamil committed:
Level-2: Add, List (corrected)

liviamil committed:
Level-3: Mark as Done

liviamil committed:
Level-4: ToDo, Event, Deadline

liviamil committed:
Modified Events and Deadline

liviamil committed:
Fixed scanner

liviamil committed:
A-TextUiTesting

liviamil committed:
Level-5: Handle Errors

liviamil committed:
Level-6: Delete

liviamil committed:
Adds Enum and fixes border printing

liviamil committed:
Fixes more printing bugs

liviamil committed:
Adds read file and save file function

liviamil committed:
Fixes bugs

liviamil committed:
Fixes NumberFormatException

liviamil committed:
Adds Dates and Times

liviamil committed:
Merges branch level-7 to master

liviamil committed:
fixes conflicts

liviamil committed:
Merges branch level-8 to master

neosunhan commented on data/Sally.txt:
```diff
@@ -0,0 +1,5 @@
+T | 1 | read book
```
Perhaps you can consider adding data files to `.gitignore`. Otherwise users who clone your repo may end up with tasks loaded that they did not add.

neosunhan commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,13 @@
+public class Event extends Task {
+    String moreInfo;
```
Missing access modifier.

neosunhan commented on src/main/java/Sally.java:
```diff
@@ -0,0 +1,291 @@
+    }
+
+    public static void readsFile() throws FileNotFoundException {
+        String filePath = "D:/NUS/Y2/S1/CS2103T/ip/data/Sally.txt";
```
Consider making `filePath` system-agnostic. Also, other users who run your app may not have a D drive.

neosunhan commented on src/main/java/Sally.java:
```diff
@@ -0,0 +1,291 @@
+    }
+
+    public static void savesFile() throws IOException {
+        String filePath = "D:/NUS/Y2/S1/CS2103T/ip/data/Sally.txt";
```
Consider storing `filePath` as a class field. This will make changing `filePath` easier as you only need to change it in one place.

neosunhan commented on src/main/java/Sally.java:
```diff
@@ -0,0 +1,291 @@
+        }
+    }
+
+    protected static String printList() {
```
Since this function does not actually do any printing, maybe you can rename it to `convertListToString` or something similar?

neosunhan reviewed (commented):
I've left some comments about the code. Regarding the storage of tasks, perhaps you can consider how other users with different machines (with different OS and file systems) will interact with your app.

liviamil committed:
Adds Storage, TaskList, Ui skeleton classes

liviamil committed:
edits classes

liviamil committed:
fixes printing

liviamil committed:
Adds More Classes

liviamil committed:
resolves Merge conflicts

liviamil committed:
organizes into Packages

liviamil committed:
Relocates Sally.java

ytingtan commented on src/main/java/sally/parser/Parser.java:
```diff
@@ -0,0 +1,134 @@
+import sally.command.UnmarkCommand;
+import sally.exception.SallyException;
+import sally.storage.Storage;
+import sally.task.*;
```
Imported packages should be listed explicitly, maybe you could write down the attributes and methods you need from the `task` package and import in one-by-one? :) Otherwise LGTM!

ytingtan commented on src/main/java/sally/parser/Parser.java:
```diff
@@ -0,0 +1,134 @@
+            }
+        }
+
+        return new ByeCommand(); // Perlu diganti
```
Oops! Gentle reminder that all comments should be written in English ^_^

ytingtan commented on src/main/java/sally/task/Task.java:
```diff
@@ -0,0 +1,67 @@
+        this.isDone = false;
+    }
+
+    public boolean getDoneStatus() {
```
As much as possible, it would be nice to name booleans to sound like booleans. Perhaps you could name this `isDone` instead? :)

ytingtan reviewed (commented):
Overall, I enjoyed reading your code as it was structured nicely and was aligned to the coding standard. I've made comments about some minor points, but overall LGTM! 👍 😄

liviamil committed:
Adds java docs

liviamil committed:
Completes class javadocs

liviamil committed:
Fixes parsing issues, tweaks some classes, and adds more javadocs

liviamil committed:
Adds Gradle Support

liviamil committed:
Adds JUnit Tests

liviamil committed:
Packages the App as a JAR file

liviamil committed:
Adds missing JavaDocs comments

liviamil committed:
Adds JavaDoc comments to Ui class

liviamil committed:
Tweaks code to comply with Java coding standard

liviamil committed:
Adds Find function

liviamil committed:
Merges branch-Level-9 to master

liviamil committed:
Merge branch 'branch-A-Codingstandard'

liviamil committed:
Adds GUI

liviamil committed:
Adds GUI (on progress)

liviamil committed:
Completes GUI

liviamil committed:
Fixes greeting message

liviamil committed:
Fixes GUI

liviamil committed:
Adds assertions

liviamil committed:
Improves code quality

liviamil committed:
Merge pull request #2 from liviamil/branch-A-Assertions

Merge branch-A-Assertions

liviamil committed:
Merge pull request #3 from liviamil/branch-A-CodeQuality

Merge branch-A-CodeQuality

liviamil committed:
Adds C-Help

liviamil committed:
Merge branch 'master' of https://github.com/liviamil/ip

liviamil committed:
Adds C-Help

liviamil committed:
Add a User Guide.

Introduce Sally's main features including examples.

liviamil committed:
Improve GUI

liviamil committed:
Tweak help command

liviamil committed:
Clean SallyException

liviamil committed:
Add Ui.png

liviamil committed:
Handle no data folder case

liviamil committed:
Add exit with bye command

liviamil committed:
Improve abstraction in Storage.java by implementing SLAP and fix checkstyle

liviamil committed:
Improve abstraction in Parser and Storage classes

- Implement SLAP to prevent deep-nested methods in Parser.java and
  Storage.java
- Fix empty todo issue to throw exception
- Fix date format in deadline display
- Add comments

liviamil committed:
Fix checkstyle

liviamil committed:
Add test cases for Parser Test

liviamil committed:
Improve Storage

liviamil committed:
Update README.md

liviamil committed:
Merge pull request #5 from liviamil/v1.2

V1.2 Improvements

liviamil committed:
Fix jar download link

liviamil committed:
Update README.md
